Sources/Profiler.swift

```diff
@@ -0,0 +1,326 @@
+//
+//  ProcedureKit
+//
+//  Copyright © 2016 ProcedureKit. All rights reserved.
+//
+
+
+import Foundation
+
+public enum ProcedureEvent: Int {
+    case attached = 0, started, cancelled, produced, finished
+}
+
+extension ProcedureEvent: CustomStringConvertible {
+    public var description: String {
+        switch self {
+        case .attached: return "Attached"
+        case .started: return "Started"
+        case .cancelled: return "Cancelled"
+        case .produced: return "Produced"
+        case .finished: return "Finished"
+        }
+    }
+}
+// MARK: - ProcedureProfilerReporter
+public protocol ProcedureProfilerReporter {
+    func finishedProfilingWithResult(result: ProfileResult)
+}
+
+// MARK: - ProfileResult
+public struct ProfileResult {
+    public let identity: Procedure.Identity
+    public let created: TimeInterval
+    public let attached: TimeInterval
+    public let started: TimeInterval
+    public let cancelled: TimeInterval?
+    public let finished: TimeInterval?
+    public let children: [ProfileResult]
+}
+
+extension PendingValue where T: Equatable {
+    var pending: Bool {
+        if case .pending = self {
+            return true
+        }
+        return false
+    }
```

Comment:
Lets take this computed property out of this extension, and add it to `PendingValue` directly in *ResultInjection.swift*, and rename it:

```swift
public enum PendingValue {
    // etc
    public var isPending: Bool {
        if case .pending = self { 
            return true
        }
    }
}
```